frameworks/elastic/universe-kibana/marathon.json.mustache

```diff
@@ -24,6 +24,7 @@
     "FRAMEWORK_NAME": "{{service.name}}",
     "FRAMEWORK_USER": "{{service.user}}",
     "KIBANA_PLUGINS": "{{kibana.plugins}}",
+    "KIBANA_PLACEMENT": "{{{kibana.placement}}}",
```

Comment:
Since Kibana is just a Marathon app and not an SDK service, just setting the environment variable won't be enough to cause the task to be scheduled based on a placement constraint. The constraint coming from the service configuration JSON needs to be injected in the `constraints` key in the Marathon app definition. There are some examples in the [Marathon constraints documentation page](https://mesosphere.github.io/marathon/docs/constraints.html).

Also, note that the Marathon placement value is an array of arrays, not a stringified array of arrays.

Reply:
I didn't get the difference between sdk service and app first, thanks for clarifying, changes in testing